sklearn/utils/_array_api.py

```diff
@@ -105,6 +105,25 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
             yield array_namespace, None, None
 
 
+def _get_namespace_device_dtype_ids(param):
+    """Get pytest parametrization IDs for `yield_namespace_device_dtype_combinations`"""
+    # See #31042 for details
+    try:
+        import array_api_strict
+    except ImportError:
+        return f"{param}"
+    else:
+        if param == array_api_strict.Device("CPU_DEVICE"):
+            return "CPU_DEVICE"
+        if param == array_api_strict.Device("device1"):
```

Comment:
You may as well implement 'device2' while you are at it (in `array-api-strict` the allowed devices are `CPU_DEVICE`, `device1`, `device2` right now in `array-api-strict`)

Reply:
Since `yield_namespace_device_dtype_combinations` never returns `array_api_strict.Device("device2")` I think the current code is fine.

Reply:
Happy to add 'device2' to future proof